src/vmm/src/io_uring/mod.rs

```diff
@@ -165,10 +165,10 @@ impl IoUring {
     /// It's up to the caller to make sure that this value is ever freed (not leaked).
     pub unsafe fn push<T>(&mut self, op: Operation<T>) -> std::result::Result<(), (Error, T)> {
         // validate that we actually did register fds
-        let fd = op.fd() as i32;
+        let fd = i32::try_from(op.fd()).unwrap();
         match self.registered_fds_count {
             0 => Err((Error::NoRegisteredFds, op.user_data())),
-            len if fd < 0 || (len as i32 - 1) < fd => {
+            len if fd < 0 || (i32::try_from(len).unwrap() - 1) < fd => {
                 Err((Error::InvalidFixedFd(fd), op.user_data()))
```

Comment:
Here we can remove casting. Only for `InvalidFixedFd` we still need to cast. But I guess it is worth fixing `InvalidFixedFd` to actually have `FixedFd` inside instead of i32.
```rust
let fd = op.fd();
match self.registered_fds_count {
     0 => Err((Error::NoRegisteredFds, op.user_data())),
     len if fd < 0 || (len - 1) < fd => {
...
```

Reply:
I changed that all to use `FixedFd` / `u32`. I also took the opportunity to clean up the second `if` test to just `fd >= len` (the `< 0` is now redundant because it's an unsigned type).